Objects/dictobject.c

```diff
@@ -1116,7 +1116,7 @@ unicodekeys_lookup_generic(PyDictObject *mp, PyDictKeysObject* dk, PyObject *key
     return do_lookup(mp, dk, key, hash, compare_unicode_generic);
 }
 
-static inline Py_ALWAYS_INLINE Py_ssize_t
+static inline Py_ssize_t
```

Comment:
Not relating to this issue, but return value should be int or bool.

Reply:
Even though this only returns `0` or `1`, it must match the signature of the `check_lookup` function pointer in `do_lookup`.

Reply:
Yes. check_lookup return type should be int instead of Py_ssize_t.

Reply:
What's the advantage of using an `int`? 

`check_lookup` may return `DKIX_ERROR` or `DKIX_KEY_CHANGED`, which we generally treat as `Py_ssize_t`. The return value from `check_lookup` is also returned unmodified by `Py_ssize_t do_lookup` in some cases.

Reply:
> What's the advantage of using an `int`?

Py_ssize_t implies it represents some sort of index or size.
By using int, we can clarify that return value doesn't represents index or size, like PyObject_RichCompareBool returns int for 1/0/-1.

Additionally, it can be efficient on platform that have 64bit address space but optimized for 32bit integer arithmetic, althouh such archtectures are very rare nowadays.

> `check_lookup` may return `DKIX_ERROR` or `DKIX_KEY_CHANGED`, which we generally treat as `Py_ssize_t`. The return value from `check_lookup` is also returned unmodified by `Py_ssize_t do_lookup` in some cases.

Theye are defined as `#define DKIX_EMPTY (-1)`. We generally use -1 for error in int and Py_ssize_t.

That is why I prefer int over Py_ssize_t here. But merit of changing it is very small so no need to fix it. "A Foolish Consistency is the Hobgoblin of Little Minds".

Reply:
Ok, I created https://github.com/python/cpython/pull/121581 to return int instead of Py_ssize_t.

Reply:
> Ok, I created https://github.com/python/cpython/pull/121581 to return int instead of Py_ssize_t.

I merged it and rebased this PR on top of it.